File: Lab1/Lab1Python/text_manager.py
def split_text_to_new_lines(string: str) -> list[str]:  # розбивання речення по рядочкам
    return string.replace(". ", ".\n").split('\n')


def define_longest_in_string(string: str) -> str:  # знайти найдовше слово, поставити на початок його і довжину
    words = string.replace(',', '').replace('.', '').split()
    max_word = ""
    len_of_max_word = 0

    for word in words:
        if len(word) > len_of_max_word:
            max_word = word
            len_of_max_word = len(word)

    return f"[{len_of_max_word} {max_word}] " + string


def define_longest_in_text(text: list[str]) -> list[str]:  # виконати завдання для всіх рядків
    result_text = []
    for line in text:
        result_text.append(define_longest_in_string(line))

    return result_text

File: Lab1/Lab1Sharp/Lab1Sharp/TextManager.cs
namespace Lab1Sharp;

internal class TextManager
{
    public string[] SplitTextToNewLines(string str) //розбивання речення по рядочкам
    {
        str = str.Replace("\r", "").Replace(".", ".\n");
        return str.Split('\n', StringSplitOptions.RemoveEmptyEntries);
    }

    public string DefineLongestInString(string str) //знайти найдовше слово, поставити на початок його і довжину
    {
        str = str.Trim(); //видалення лишніх пробілів зліва і справа
        string[] words = str.Split(new char[] {' ', ',', '.'}, StringSplitOptions.RemoveEmptyEntries);
        string maxWord = words.First();
        int lenOfMaxWord = maxWord.Length;

        foreach (string word in words)
        {
            if (word.Length > lenOfMaxWord)
            {
                maxWord = word;
                lenOfMaxWord = word.Length;
            }
        }

        return $"[{lenOfMaxWord} {maxWord}] {str}";
    }

    public string[] DefineLongestInText(string[] text) //виконати завдання для всіх рядків
    {
        string[] resultText = new string[text.Length];

        for (int i = 0; i < text.Length; i++)
        {
            resultText[i] = DefineLongestInString(text[i]);
        }

        return resultText;
    }
}